From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david@flypig.co.uk>
Date: Fri, 1 Dec 2023 23:56:36 +0000
Subject: [PATCH] [sailfishos][gecko] Allow file scheme when loading OpenSearch
 providers

When loading OpenSearch provider XML files the scheme was restricted to
https. This prevented the OpenSearch XML provider files from being
loaded from disk at initialisation or when installing a search provider.

This change extends the support to include both https and file schemes
so that the providers can be loaded correctly from disc.
---
 toolkit/components/search/OpenSearchEngine.jsm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/toolkit/components/search/OpenSearchEngine.jsm b/toolkit/components/search/OpenSearchEngine.jsm
index a774de71b064..fe064d752e8f 100644
--- a/toolkit/components/search/OpenSearchEngine.jsm
+++ b/toolkit/components/search/OpenSearchEngine.jsm
@@ -91,7 +91,7 @@ class OpenSearchEngine extends SearchEngine {
         Cr.NS_ERROR_UNEXPECTED
       );
     }
-    if (!/^https?$/i.test(loadURI.scheme)) {
+    if (!/^(?:https?|file)$/i.test(loadURI.scheme)) {
       throw Components.Exception(
         "Invalid URI passed to SearchEngine constructor",
         Cr.NS_ERROR_INVALID_ARG
